packages/next/lib/is-error.ts

```diff
@@ -11,3 +13,22 @@ export default function isError(err: unknown): err is NextError {
     typeof err === 'object' && err !== null && 'name' in err && 'message' in err
   )
 }
+
+export function getProperError(err: unknown): Error {
+  if (isError(err)) {
+    return err
+  }
+
+  if (process.env.NODE_ENV === 'development') {
+    // provide better error for case where `throw undefined`
+    // is called in development
+    if (typeof err === 'undefined') {
+      return new Error(
+        'An undefined error was thrown, ' +
+          'see here for more info: https://nextjs.org/docs/messages/threw-undefined'
+      )
```

Comment:
Would you be able to to include the stringify version of the undefined error here as well? The original reason I went for the PR was because of how much time I spent digging through the code to figure out what the error was and it turned out to be thrown from another 3rd party library. Just making sure it doesn't go from `"[Object object]"` to `"An undefined error was thrown"` during development.

Reply:
This is for the specific case where `throw undefined` occurs stringifying would just show `new Error("undefined")`. The test case you added is still present and passing and `throw { hello: 'world' }` will be stringified still a couple lines down.

Reply:
Oh perfect, I missed that.